api/src/client.rs

```diff
@@ -40,19 +40,16 @@ where
 /// object as body on a given URL that returns a JSON object. Handles request
 /// building, JSON serialization and deserialization, and response code
 /// checking.
-pub fn post<'a, IN, OUT>(url: &'a str, input: &IN) -> Result<OUT, Error>
+pub fn post<'a, IN>(url: &'a str, input: &IN) -> Result<(), Error>
```

Comment:
There's a mismatch right now between the old style Api stuff and the newer Iron handlers.
We don't have any api clients that expect anything but `()` as a response so I have simplified this for now.
We were seeing issues parsing empty responses from Iron handlers here.